sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -71,7 +71,7 @@ def _need_new_token(self):
         return not self._token or self._token.expires_on - time.time() < 300
 
 
-class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, SansIOHTTPPolicy):
+class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, SansIOHTTPPolicy, HTTPPolicy):
```

Comment:
The the pipeline will call on_request or send or both? 

Do we have a design for this scenario? @annatisch

Reply:
After #16726, the pipeline will call this policy's send method (instead of _SansIOHTTPPolicyRunner.send). Another option is for this policy not to inherit SansIOHTTPPolicy.

Reply:
Since this is an "AvecIOHTTPPolicy" (as opposed to Sans), I would remove the inheritance from `SansIOHTTPPolicy` and derive from the "correct" base class. I think that your implementation can call `self.on_request` and `self.on_response` to make sure they are called in a very similar fashion as before in order to handle any cases where someone has extended/derived from the policy, no?

Reply:
Have done. The extended policy follows _SansIOHTTPPolicyRunner's calling pattern such that it calls a subclass's `on_*` as the runner would.